alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt

```diff
@@ -554,9 +609,12 @@ class AlertService(
                     }
                 }
                 Alert.State.AUDIT -> {
+                    val index = if (alertIndices.isAlertHistoryEnabled()) {
+                        dataSources.alertsHistoryIndex
```

Comment:
Can flipping the alert history index enablement hinder the ability to read the history data during upgrades?

Reply:
yes that problem exists. we have an issue created to deal with it